vnext/Desktop.UnitTests/MemoryMappedBufferTests.cpp

```diff
@@ -0,0 +1,164 @@
+#include "MemoryMappedBuffer.h"
+#include "utilities.h"
+
+#include <CppUnitTest.h>
+
+#include <shlwapi.h>
+#include <windows.h>
+
+#include <cstring>
+#include <memory>
+
+using facebook::jsi::Buffer;
+using facebook::jsi::JSINativeException;
+using facebook::react::utilities::checkedReinterpretCast;
+using Microsoft::JSI::MakeMemoryMappedBuffer;
+using Microsoft::VisualStudio::CppUnitTestFramework::Assert;
+
+namespace {
+
+std::wstring GetTestFileName() {
+  wchar_t tempPath[MAX_PATH];
+  if (!GetTempPathW(MAX_PATH, tempPath)) {
+    std::terminate();
+  }
+
+  wchar_t testFileName[MAX_PATH];
+  if (!GetTempFileNameW(
+          tempPath, L"MemoryMappedBufferUnitTests", 0, testFileName)) {
+    std::terminate();
+  }
+
+  return std::wstring(testFileName);
+}
+
+uint32_t GetPageSize() noexcept {
+  SYSTEM_INFO systemInfo;
+  GetSystemInfo(&systemInfo);
+  return systemInfo.dwPageSize;
+}
+
+} // anonymous namespace
+
+TEST_CLASS(MemoryMappedBufferUnitTests) {
+ private:
+  std::wstring m_testFileName;
+
+  void WriteTestFile(const char *const content, size_t size) {
+    FILE *testFilePtr;
+
+    // We need pass in the "b" flag to ensure that fwrite modify its input when
+    // outputting to disk.
+    if (_wfopen_s(&testFilePtr, m_testFileName.c_str(), L"wb")) {
+      std::terminate();
```

Comment:
>std::terminate(); [](start = 6, length = 17)

maybe this is not related to this particular CR. The problem is that std::terminate doesn't produce Watson dump containing callstack AFAIK. So investigating such failures will be very difficult, especially if they are intermittent and fail on Continuous Integration machines.

Either VEC or throwing C++ exception produces usable dump. You can leave this as is for this test, but we should address this at some point.

Reply:
I agree this warrants a discussion.

---
In reply to: [321859108](https://github.com/microsoft/react-native-windows/pull/3059#discussion_r321859108) [](ancestors = 321859108)